semgrep-core/src/core/Rule.ml

```diff
@@ -167,15 +167,30 @@ type pformula = New of formula | Old of formula_old [@@deriving show, eq]
 (* Taint-specific types *)
 (*****************************************************************************)
 
-type sanitizer_spec = {
+type taint_source = {
+  formula : pformula;
+  label : string;  (** The label to attach to the data *)
+  requires : AST_generic.expr;
+      (** A Boolean formula over taint labels, the expression that
```

Comment:
please do not use "formula" as we already use it for the formula/combination of patterns.